### PR TITLE
Convert from Kubernetes Endpoints to EndpointSlices

### LINKED ISSUE #7: Renaming/deleting ports causes an inconsistent state
We never get an endpoint-specific delete event for this case, so the endpoint with the old name persists.

### PR DESCRIPTION
**Pull request by abursavich**
I'm not sure if you're interested in maintaining this repo, but I needed this change and wanted to try to give it back.

This replaces Endpoints with [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/), which have been stable since Kubernetes v1.21 and reduce load on the Kubernetes API Server.  I exceeded the [capacity limit of Endpoints](https://kubernetes.io/docs/concepts/services-networking/service/#over-capacity-endpoints), which caused a bunch of problems.

This also upgrades Go, module dependencies, and grpc-health-probe.

As an extra benefit, it also fixes #7.

### CODE REVIEW

File: deploy/base/clusterrole.yaml [hunk 1]
```diff
@@ -4,11 +4,8 @@ metadata:
     name: ekglue-reader
 rules:
     - apiGroups: [""]
-      resources: ["services"]
+      resources: ["nodes"", "services"]
```
Comment (jrockway) [inline deploy/base/clusterrole.yaml:7]: Extra quote?

**Review by jrockway — APPROVED** [review]
I wouldn't say this project is unmaintained, just that I've had nothing to add recently.  Thank you so much for your contribution!

File: deploy/base/clusterrole.yaml (hunk 1 above)
Comment (abursavich) [reply deploy/base/clusterrole.yaml]: Fixed.

**Comment by jrockway** [comment]
I appreciate the contribution!  Thank you so much!

**Comment by abursavich** [comment]
> I appreciate the contribution! Thank you so much!

No problem. Thanks for making it available in the first place.

**Pull request merged**